### PR TITLE
Avoid rate limit exceeded exceptions when sending metrics

### PR DESCRIPTION
**Pull request by alicefuzier**
🍦 🍰

### CODE REVIEW

File: common/src/main/scala/uk/ac/wellcome/metrics/MetricsSender.scala [hunk 1]
```diff
@@ -15,8 +18,31 @@ import scala.concurrent.duration._
...
 class MetricsSender @Inject()(@Flag("aws.metrics.namespace") namespace: String,
-                              amazonCloudWatch: AmazonCloudWatch)
+                              amazonCloudWatch: AmazonCloudWatch, actorSystem: ActorSystem)
     extends Logging {
+  implicit val system = actorSystem
+  implicit val materialiser = ActorMaterializer()
+
+  // According to https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_limits.html
+  // PutMetricData supports at maximum 20 MetricDatum per PutMetricDataRequest.
+  // The maximum number of PutMetricData requests is 150 per second.
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/metrics/MetricsSender.scala]: 👌 

Nice comments!

File: common/src/main/scala/uk/ac/wellcome/metrics/MetricsSender.scala [hunk 2]
```diff
@@ -15,8 +18,31 @@ import scala.concurrent.duration._
...
 
 class MetricsSender @Inject()(@Flag("aws.metrics.namespace") namespace: String,
-                              amazonCloudWatch: AmazonCloudWatch)
+                              amazonCloudWatch: AmazonCloudWatch, actorSystem: ActorSystem)
     extends Logging {
+  implicit val system = actorSystem
+  implicit val materialiser = ActorMaterializer()
+
+  // According to https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_limits.html
+  // PutMetricData supports at maximum 20 MetricDatum per PutMetricDataRequest.
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/metrics/MetricsSender.scala]: nit: “*a* maximum”?

**Comment by alexwlchan** [comment]
Resolves #644.

File: common/src/main/scala/uk/ac/wellcome/metrics/MetricsSender.scala [hunk 3]
```diff
@@ -15,8 +23,39 @@ import scala.concurrent.duration._
...
+  // The maximum number of PutMetricData requests is 150 per second.
+  private val metricDataListMaxSize = 20
+  private val maxPutMetricDataRequestsPerSecond = 150
+
+  val sourceQueue: SourceQueueWithComplete[MetricDatum] =
+    Source
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/metrics/MetricsSender.scala]: 👍

**Pull request merged**